src/fastertransformer/triton_backend/deberta/DebertaTritonModelInstance.cc

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/fastertransformer/triton_backend/deberta/DebertaTritonModelInstance.h"
+#include "src/fastertransformer/triton_backend/triton_utils.hpp"
+
+namespace ft = fastertransformer;
+
+template<typename T>
+void triton_stream_callback(std::unordered_map<std::string, ft::Tensor>* output_tensors, void* ctx)
+{
+    DebertaTritonModelInstance<T>* model  = reinterpret_cast<DebertaTritonModelInstance<T>*>(ctx);
+    auto                        result = DebertaTritonModelInstance<T>::convert_outputs(*output_tensors);
+
+    model->stream_cb_(result, model->stream_ctx_);
+}
+
+template<typename T>
+DebertaTritonModelInstance<T>::DebertaTritonModelInstance(std::unique_ptr<ft::Deberta<T>>                       deberta,
+                                                    std::shared_ptr<ft::DebertaWeight<T>>                       deberta_weight,
+                                                    std::unique_ptr<ft::Allocator<ft::AllocatorType::CUDA>>     allocator,
+                                                    std::unique_ptr<ft::cublasAlgoMap>                          cublas_algo_map,
+                                                    std::unique_ptr<std::mutex>                                 cublas_wrapper_mutex,
+                                                    std::unique_ptr<ft::cublasMMWrapper>                        cublas_wrapper,
+                                                    std::unique_ptr<cudaDeviceProp>                             cuda_device_prop_ptr):
+    deberta_(std::move(deberta)),
+    deberta_weight_(deberta_weight),
+    allocator_(std::move(allocator)),
+    cublas_algo_map_(std::move(cublas_algo_map)),
+    cublas_wrapper_mutex_(std::move(cublas_wrapper_mutex)),
+    cublas_wrapper_(std::move(cublas_wrapper)),
+    cuda_device_prop_ptr_(std::move(cuda_device_prop_ptr))
+{
+}
+
+template<typename T>
+std::shared_ptr<std::vector<triton::Tensor>>
+DebertaTritonModelInstance<T>::forward(std::shared_ptr<std::vector<triton::Tensor>> input_tensors)
```

Comment:
this is not used by triton, so didn't implement it as the bert implementation.

Reply:
Is there a way to print out a descriptive message if this is exercised?

Reply:
don't know, i basically follow the convention of how bert is implemented.